server-worker/lines/nz-wlg.js

```diff
@@ -15,210 +15,124 @@ const regionEnum = {
+    items: [],
+  },
+  {
+    name: 'Hutt Valley',
+    items: [],
+  },
+  {
+    name: 'Porirua',
+    items: [],
+  },
+  {
+    name: 'Paraparaumu',
+    items: [],
+  },
+  {
+    name: 'Wairarapa',
+    items: [],
+  },
+  {
+    name: 'Late Night',
+    items: [],
+  },
 ]
 const friendlyNames = {}
 const allLines = {}
 const lineOperators = {}
 
-const sortLines = (lines) => {
-  const sortFn = (a, b) => {
-    const parsedA = parseInt(a.replace(/\D+/g, ''))
-    const parsedB = parseInt(b.replace(/\D+/g, ''))
+const sortLines = lineGroups => {
+  lineGroups.forEach(group => {
+    group.items.sort((a, b) => {
+      const parsedA = parseInt(a.replace(/\D+/g, ''))
+      const parsedB = parseInt(b.replace(/\D+/g, ''))
 
-    if (parsedA === parsedB) {
-      return a.length - b.length
-    }
-    return parsedA - parsedB
-  }
-  lines.forEach((i) => {
-    i.items.sort(sortFn)
+      if (parsedA === parsedB) {
+        return a.length - b.length
+      }
+      return parsedA - parsedB
+    })
   })
 }
 
 const getLines = () => {
   const sqlRequest = connection.get().request()
-  sqlRequest.query(`
+  sqlRequest
+    .query(
+      `
     SELECT
-      route_short_name, route_long_name, agency_id, route_type
+      route_short_name, route_long_name, agency_id, route_type, route_color
     FROM routes
     ORDER BY route_type, route_short_name
-  `).then(result => {
-    result.recordset.forEach((record) => {
-      allLines[record.route_short_name] = [[record.route_long_name]]
+  `
+    )
+    .then(result => {
+      result.recordset.forEach(record => {
+        allLines[record.route_short_name] = [[record.route_long_name]]
 
-      lineOperators[record.route_short_name] = record.agency_id
-      if (record.route_type !== 3) {
-        lineGroups[regionEnum.cf].items.push(record.route_short_name)
-      } else {
-        if (record.route_short_name.slice(0,1) === 'N') {
-          lineGroups[regionEnum.late].items.push(record.route_short_name)
-        } else if (parseInt(record.route_short_name) >= 250) {
-          lineGroups[regionEnum.paraparaumu].items.push(record.route_short_name)
-        } else if (parseInt(record.route_short_name) >= 200 && parseInt(record.route_short_name) < 210) {
-          lineGroups[regionEnum.wairarapa].items.push(record.route_short_name)
-        } else if (parseInt(record.route_short_name) >= 80 && parseInt(record.route_short_name) < 200) {
-          lineGroups[regionEnum.hutt].items.push(record.route_short_name)
-        } else if (parseInt(record.route_short_name) > 200) {
-          lineGroups[regionEnum.porirua].items.push(record.route_short_name)
+        lineOperators[record.route_short_name] = record.agency_id
+        if (record.route_type !== 3) {
+          lineGroups[regionEnum.cf].items.push(record.route_short_name)
         } else {
-          lineGroups[regionEnum.central].items.push(record.route_short_name)
+          if (record.route_short_name.slice(0, 1) === 'N') {
```

Comment:
would like some line comments to know what you're trying to achieve by this slice

Reply:
Checks if the first character is N?

Reply:
y tho